config/config.go

```diff
@@ -939,6 +944,9 @@ type ExpertConfig struct {
 	// TestGossipProbeInterval defines the probe interval used by the gossip
 	// service in tests.
 	TestGossipProbeInterval time.Duration
+	// NodeRegistryFactory defines a custom node registry function that can be used
+	// instead of a static registry or the built in memberlist gossip mechanism.
+	NodeRegistryFactory NodeRegistryFactory
```

Comment:
As this PR will introduce a new mode of node registry, I think the godoc of the registry feature should be updated to briefly describe this new mode. Currently, the godoc of the NodeHostConfig.GossipConfig field describes the node registry feature itself, maybe just add some description there to point out such two modes.